PythonVisualizations/SimpleQueue.py

```diff
@@ -8,60 +8,85 @@
 
 WIDTH = 800
 HEIGHT = 400
-
 CELL_SIZE = 50
 ARRAY_X0 = 100
 ARRAY_Y0 = 100
 
-class Queue(object):
+class Queue(VisualizationApp):
 
     colors = ['red', 'green', 'blue', 'orange', 'yellow', 'cyan', 'magenta',
               'dodgerblue', 'turquoise', 'grey', 'gold', 'pink']
     nextColor = 0
 
-    def __init__(self, size=12):
+    def __init__(self, size=12, title="Queue", **kwargs):
+        super().__init__(title=title, **kwargs)
         self.list = [None]*size
         self.size = size
         self.front = 1  # when Queue is empty, front 
         self.rear = 0   # should be to right of rear.
         self.nItems = 0
-        
+        self.rearArrow = (None, None)
+        self.frontArrow = (None, None)
+        self.buttons = self.makeButtons()
+        self.display()
+
     def __str__(self):
         return str(self.list)
     
     # ANIMATION METHODS
-    def speed(self, sleepTime):
-        return (sleepTime * (scaleDefault + 50)) / (scale.get() + 50)
+    def createIndex(  # Create an index arrow to point at an indexed
+            self, index=-1, name=None, high=True):  # cell with an optional name label
+
+        coords = [ARRAY_X0 + CELL_SIZE * index, ARRAY_Y0, ARRAY_X0 + CELL_SIZE * (index + 1), ARRAY_Y0 + CELL_SIZE]
+
+        x0 = coords[0]
+        x1 = coords[0]
+        y0 = coords[1] - CELL_SIZE * 2 // 2
+        y1 = coords[1] - CELL_SIZE * 1 // 4
+        if high:  # changes height of arrow
+            y0 -= 20
+
+        arrow = self.canvas.create_line(
+            x0, y0, x1, y1, arrow="last", fill=self.VARIABLE_COLOR)
```

Comment:
In the other visualizations, the pointer variables point at the center of the array cell.  I think we should do the same for the Queue,  so
```
        x0 = coords[0] + self.CELL_SIZE // 2
        x1 = x0
```